From 6309f331a6ca3d7e82c91bbf6af7a515cef2fdcd Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Tue, 22 Sep 2020 12:24:10 -0700
Subject: [PATCH 1634/1851] v4l: xilinx: tpg: Fix warnings for CHECKED_RETURN

commit ce6fcaae123bf634eac6e41a587b24cf96d67a86 from
https://github.com/Xilinx/linux-xlnx.git

This patch fix the warnings for the CHECKED_RETURN type.

The actual warning is:

For xtpg_s_stream():
Event check_return: 	Calling "v4l2_ctrl_handler_setup" without
checking return value (as is done elsewhere 7 out of 8 times).

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-tpg.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/media/platform/xilinx/xilinx-tpg.c b/drivers/media/platform/xilinx/xilinx-tpg.c
index 051025d319fc..8c0d6937f7ba 100644
--- a/drivers/media/platform/xilinx/xilinx-tpg.c
+++ b/drivers/media/platform/xilinx/xilinx-tpg.c
@@ -284,6 +284,7 @@ static int xtpg_s_stream(struct v4l2_subdev *subdev, int enable)
 		if (!xtpg->is_hls) {
 			xvip_stop(&xtpg->xvip);
 		} else {
+			int ret;
 			/*
 			 * There is an known issue in TPG v7.0 that on
 			 * resolution change it doesn't generates pattern
@@ -292,7 +293,13 @@ static int xtpg_s_stream(struct v4l2_subdev *subdev, int enable)
 			 */
 			gpiod_set_value_cansleep(xtpg->rst_gpio, 0x1);
 			gpiod_set_value_cansleep(xtpg->rst_gpio, 0x0);
-			v4l2_ctrl_handler_setup(&xtpg->ctrl_handler);
+			ret = v4l2_ctrl_handler_setup(&xtpg->ctrl_handler);
+			if (ret) {
+				struct device *dev = xtpg->xvip.dev;
+
+				dev_err(dev, "failed to set controls\n");
+				return ret;
+			}
 		}
 
 		if (xtpg->vtc)
-- 
2.31.1

